bundler/lib/bundler/cli/install.rb

```diff
@@ -66,7 +66,7 @@ def run
 
       Plugin.gemfile_install(Bundler.default_gemfile) if Bundler.feature_flag.plugins?
 
-      definition = Bundler.definition
+      definition = Bundler.definition(validate_dependencies: true)
```

Comment:
The `unlock` hash is supposed to have "things to be potentially unlocked" as keys, like `:gems` (this list of gems), `:sources` (this list of sources), `:bundler` (the locked version of Bundler) or `:ruby` (the locked version of ruby).

Hence my previous suggestion of `:lockfile_dependencies_inconsistent_with_installed_specs` as just another thing the can be potentially unlocked. I understand it's really verbose, maybe `:dependencies_inconsistent_with_installed_specs` then, because the `lockfile_` part is redundant (if you're unlocking something, that kind of implies that thing is locked).

Reply:
Thanks, this explanation was helpful.